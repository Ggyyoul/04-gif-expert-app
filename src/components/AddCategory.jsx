import { useState } from "react";
import "./AddCategory.css";
export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("Hola Mundo");

  const hableInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmite = e => {
    e.preventDefault();
  }

  return (
    <form onSubmit={handleSubmite}>
      <input
        type="text"
        placeholder="Add Category"
        value={inputValue}
        onChange={hableInputChange}
      />
    </form>
  );
};
