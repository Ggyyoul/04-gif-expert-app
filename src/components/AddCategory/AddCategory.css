input {
  width: 100%;
  font-family:fantasy;
  font-size: 1.2rem;
  color: slateblue;
}