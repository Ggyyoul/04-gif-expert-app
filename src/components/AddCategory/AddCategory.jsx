import { useState } from "react";
import PropTypes from 'prop-types';
import "./AddCategory.css";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const hableInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmite = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories( categories => [...categories, inputValue])
      setInputValue("")
    }
  };

  return (
    <form onSubmit={handleSubmite}>
      <input
        type="text"
        placeholder="Add Category"
        value={inputValue}
        onChange={hableInputChange}
      />
    </form>
  );
};


AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}