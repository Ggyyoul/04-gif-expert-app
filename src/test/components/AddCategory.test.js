import { AddCategory } from '../../components/AddCategory/AddCategory.jsx';
const { shallow } = require('enzyme');

describe('tests in componenet <AddCategory />', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
    //Limpia lo mocks despues de realizar cada prueba a su valor inicial
    jest.clearAllMocks();
  })


  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'hola Mundo';
    //entre llaves esta el valor del evento 'e'
    input.simulate('change', {
      target: {
        value: value,
      },
    });

    expect(wrapper.find('p').text().trim()).toBe(value)
  });

  test('should not send info when call submit', () => {
    wrapper.find('form').simulate('submit', {preventDefault(){} })
    expect(setCategories).not.toHaveBeenCalled()
  })
  
  test('should call setCategories and clean textbox', () => {

    const value = 'Twice'
    wrapper.find('input').simulate('change',{target:{value}})
    wrapper.find('form').simulate('submit', {preventDefault(){}})
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toBeCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith( expect.any(Function))
    expect(wrapper.find('input').prop('value')).toBe('')
  })
  
});
