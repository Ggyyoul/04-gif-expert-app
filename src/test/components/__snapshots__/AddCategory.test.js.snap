// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in componenet <AddCategory /> should render component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Add Category"
    type="text"
    value=""
  />
</form>
`;
